deployer/k8s_deployer.go

```diff
@@ -0,0 +1,568 @@
+package deployer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net"
+	"regexp"
+
+	"github.com/pkg/errors"
+	client "github.com/threefoldtech/grid3-go/node"
+	"github.com/threefoldtech/grid3-go/subi"
+	"github.com/threefoldtech/grid3-go/workloads"
+	wl "github.com/threefoldtech/grid3-go/workloads"
+	"github.com/threefoldtech/substrate-client"
+	"github.com/threefoldtech/zos/pkg/gridtypes"
+	"github.com/threefoldtech/zos/pkg/gridtypes/zos"
+)
+
+type K8sDeployer struct {
```

Comment:
deployer will no longer handle the k8s. we need to pass it through deploy and cancel functions